cpp/include/raft/sparse/matrix/preprocessing.cuh

```diff
@@ -0,0 +1,143 @@
+ * @param values: Input COO values array
+ * @param values_out: Output COO values array
+ * @param n_rows: Number of rows in matrix
+ * @param n_cols: Number of columns in matrix
+ * @param k_param: K value to use for BM25 algorithm
+ * @param b_param: B value to use for BM25 algorithm
+ */
+template <typename T1, typename T2, typename IdxT>
+void encode_bm25(raft::resources& handle,
+                 raft::device_vector_view<T1, IdxT> rows,
+                 raft::device_vector_view<T1, IdxT> columns,
+                 raft::device_vector_view<T2, IdxT> values,
+                 raft::device_vector_view<T2, IdxT> values_out,
+                 int n_rows,
+                 int n_cols,
+                 float k_param = 1.6f,
+                 float b_param = 0.75)
+{
+  return matrix::detail::base_encode_bm25<T1, T2, IdxT>(
+    handle, rows, columns, values, values_out, n_rows, n_cols, k_param, b_param);
+}
+
+/**
+ * @brief Use BM25 algorithm to encode features in COO sparse matrix
+ * @param handle: raft resource handle
+ * @param coo_in: Input COO matrix
+ * @param values_out: Output values array
+ * @param k_param: K value to use for BM25 algorithm
+ * @param b_param: B value to use for BM25 algorithm
+ */
+template <typename T1, typename T2, typename IdxT>
+void encode_bm25(raft::resources& handle,
+                 raft::device_coo_matrix_view<T2, T1, T1, T1> coo_in,
+                 raft::device_vector_view<T2, IdxT> values_out,
+                 float k_param = 1.6f,
+                 float b_param = 0.75)
+{
+  return matrix::detail::encode_bm25<T1, T2, IdxT>(handle, coo_in, values_out, k_param, b_param);
+}
+
+/**
+ * @brief Use BM25 algorithm to encode features in CSR sparse matrix
+ * @param handle: raft resource handle
+ * @param csr_in: Input CSR matrix
+ * @param values_out: Output values array
+ * @param k_param: K value to use for BM25 algorithm
+ * @param b_param: B value to use for BM25 algorithm
+ */
+template <typename T1, typename T2, typename IdxT>
+void encode_bm25(raft::resources& handle,
+                 raft::device_csr_matrix_view<T2, T1, T1, T1> csr_in,
+                 raft::device_vector_view<T2, IdxT> values_out,
+                 float k_param = 1.6f,
+                 float b_param = 0.75)
+{
+  return matrix::detail::encode_bm25<T1, T2, IdxT>(handle, csr_in, values_out, k_param, b_param);
+}
+
+/**
+ * @brief Use TFIDF algorithm to encode features in COO sparse matrix
+ * @param handle: raft resource handle
+ * @param rows: Input COO rows array
+ * @param columns: Input COO columns array
+ * @param values: Input COO values array
+ * @param values_out: Output COO values array
+ * @param n_rows: Number of rows in matrix
+ * @param n_cols: Number of columns in matrix
+ */
+template <typename T1, typename T2, typename IdxT>
+void encode_tfidf(raft::resources& handle,
+                  raft::device_vector_view<T1, IdxT> rows,
+                  raft::device_vector_view<T1, IdxT> columns,
```

Comment:
This one too- if we already support the sparse formats, we shouldn't be accepting raw arrays.